habitat_sim/simulator.py

```diff
@@ -294,8 +270,13 @@ def add_sensor(
         if agent_id is None:
             agent_id = self._default_agent_id
         agent = self.get_agent(agent_id=agent_id)
-        agent._add_sensor(sensor_spec)
-        self._update_simulator_sensors(sensor_spec.uuid, agent_id=agent_id)
+        agent._add_sensor(sensor_spec, modify_agent_config=False)
+        self._sensor_readers[sensor_spec.uuid] = SensorReader(
```

Comment:
Wait, what is the `_sensor_readers[sensor_spec.uuid]`? 
May I understand what is the purpose to have such a `SensorReader` class?
At a glance, this `_sensor_readers` serves as a global dict again to me, which is a RED flag.
The main purpose of this PR is to remove such kind of global lookup table, hiding them in the C++ library (e.g., we move and manage the SensorSuite into the node class), what you had here is doing it in the opposite way. 

Again, I might be wrong. But I would like you to justify such design.

Reply:
To explain what I'm doing here:
I removed the global dict _sensors in simulator as now sensors are being kept track of by the SensorSuites. With this I also removed any code to update _sensors, including `update_simulator_sensors`, which initializes the Sensor wrapper.

Python has its own implementation to get observations, which requires initializing a Sensor wrapper which contains torch/numpy buffer and noise model for sensors and storing it in agent_observations. 

In order to make the Sensor wrapper more intuitive to understand, I renamed it to SensorReader, because all it really does is initialize the buffer and noise model and provide its own implementation to get observations. Then we needed a _sensor_readers global dict to hold these buffers and noise models, so functionally nothing has really changed.

However, I realize this is not the approach we want to go for. We really don't want this global dict.